src/common/src/types/mod.rs

```diff
@@ -1049,12 +1065,27 @@ mod tests {
 
     #[test]
     fn test_size() {
-        assert_eq!(std::mem::size_of::<StructValue>(), 16);
-        assert_eq!(std::mem::size_of::<ListValue>(), 16);
+        use static_assertions::const_assert_eq;
+
+        use crate::array::*;
+
+        macro_rules! assert_item_size_eq {
+            ($array:ty, $size:literal) => {
+                const_assert_eq!(std::mem::size_of::<<$array as Array>::OwnedItem>(), $size);
+            };
+        }
+
+        assert_item_size_eq!(StructArray, 16); // Box<[Datum]>
+        assert_item_size_eq!(ListArray, 16); // Box<[Datum]>
+        assert_item_size_eq!(Utf8Array, 16); // Box<str>
+        assert_item_size_eq!(IntervalArray, 16);
+        assert_item_size_eq!(NaiveDateTimeArray, 12);
+
         // TODO: try to reduce the memory usage of `Decimal`, `ScalarImpl` and `Datum`.
-        assert_eq!(std::mem::size_of::<Decimal>(), 20);
-        assert_eq!(std::mem::size_of::<ScalarImpl>(), 32);
-        assert_eq!(std::mem::size_of::<Datum>(), 32);
+        assert_item_size_eq!(DecimalArray, 20);
+
+        const_assert_eq!(std::mem::size_of::<ScalarImpl>(), 24);
+        const_assert_eq!(std::mem::size_of::<Datum>(), 24);
```

Comment:
I get error for this part(after cargo clean) :
```
error[E0080]: evaluation of constant value failed
    --> src/common/src/types/mod.rs:1102:9
     |
1102 |         const_assert_eq!(std::mem::size_of::<ScalarImpl>(), 24);
     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ attempt to compute `0_usize - 1_usize`, which would overflow
     |
     = note: this error originates in the macro `const_assert` which comes from the expansion of the macro `const_assert_eq` (in Nightly builds, run with -Z macro-backtrace for more info)
```

But assert_eq is ok. 

Is it my local toolchain version problem?

Reply:
Do you use `Vec` for owned item in `ByteaArray`? If so, since 8 bytes of capacity is wasted, it will become 32 again. 🤔

Reply:
Yes indeed, it should become 32 again to make #6426 pass 😢

Reply:
I find that I forgot to submit a comment. 🤣